src/algorithms.rs

```diff
@@ -731,4 +870,138 @@ mod algorithm_tests {
         assert!(&a != &b);
         assert_ne!(&a, &b);
     }
+
+    #[test]
+    fn test_extended_gcd_example() {
+        // simple example for wikipedia
+        let a = BigUint::from_u32(240).unwrap();
+        let b = BigUint::from_u32(46).unwrap();
+        let (q, s_k, t_k) = extended_gcd(&a, &b);
+
+        assert_eq!(q, BigInt::from_i32(2).unwrap());
+        assert_eq!(s_k, BigInt::from_i32(-9).unwrap());
+        assert_eq!(t_k, BigInt::from_i32(47).unwrap());
+    }
+
+
+
+    #[test]
+    fn test_jacobi() {
+        let cases = [
+            [0, 1, 1],
+            [0, -1, 1],
+            [1, 1, 1],
+            [1, -1, 1],
+            [0, 5, 0],
+            [1, 5, 1],
+            [2, 5, -1],
+            [-2, 5, -1],
+            [2, -5, -1],
+            [-2, -5, 1],
+            [3, 5, -1],
+            [5, 5, 0],
+            [-5, 5, 0],
+            [6, 5, 1],
+            [6, -5, 1],
+            [-6, 5, 1],
+            [-6, -5, -1],
+        ];
+
+        for case in cases.iter() {
+            let x = BigInt::from_i64(case[0]).unwrap();
+            let y = BigInt::from_i64(case[1]).unwrap();
+
+            assert_eq!(case[2] as isize, jacobi(&x, &y), "jacobi({}, {})", x, y);
+        }
+    }
+
+
+
+    #[test]
+    fn test_mod_inverse() {
+        let tests = [
+            ["1234567", "458948883992"],
+	    ["239487239847", "2410312426921032588552076022197566074856950548502459942654116941958108831682612228890093858261341614673227141477904012196503648957050582631942730706805009223062734745341073406696246014589361659774041027169249453200378729434170325843778659198143763193776859869524088940195577346119843545301547043747207749969763750084308926339295559968882457872412993810129130294592999947926365264059284647209730384947211681434464714438488520940127459844288859336526896320919633919"],
+	    ["-10", "13"],
+            ["-6193420858199668535", "2881"],
```

Comment:
formatting seems off, maybe just run `cargo fmt` on all the files

Reply:
Okie done